rfcs/text/0011_global_search.md

```diff
@@ -0,0 +1,629 @@
+the `GlobalSearchFindOptions.aborted$` option and then emitting from it.
+
+When this observable is provided and emitting, the GS service will complete the result observable.
+
+This observable will also be passed down to the underlying result providers, that can leverage it to cancel any pending
+asynchronous task and perform cleanup if necessary.
+
+# Drawbacks
+
+See alternatives.
+
+# Alternatives
+
+## Result providers could be only registrable from the server-side API
+
+The fact that some kinds of results, and therefore some result providers, must be on the client-side makes the API more complex,
+while making these results not available from the server-side and HTTP APIs.
+
+We could decide to only allow providers registration from the server-side. It would reduce API exposure, while simplifying
+the service implementation. However to do that, we would need to find a solution to be able to implement a server-side
+result provider for `application` (and later `management_section`) type provider.
+
+I will directly exclude the option to move the `application` registration (`core.application.register`) from client
+to server-side, as it's a very heavy impacting (and breaking) change to `core` APIs that would requires more reasons
+than just this RFC/API to consider.
+
+### AST parsing
+
+One option to make the `application` results 'visible' from the server-side would be to parse the client code at build time
+using AST to find all usages to `application.register` inspect the parameters, and generates a server file
+containing the applications. The server-side `application` result provider would then just read this file and uses it
+to return application results.
+
+However
+
+- At the parsing would be done at build time, we would not be able to generate entries for any 3rd party plugins
+- As entries for every existing applications would be generated, the search provider would to be able to know which
+  applications are actually enabled/accessible at runtime to filter them, which is all but easy
+- It will also not contains test plugin apps, making it really hard to FTR
+- AST parsing is a complex mechanism for an already unsatisfactory alternative
+
+### Duplicated server-side `application.register` API
+
+One other option would be to duplicate the `application.register` API on the server side, with a subset of the
+client-side metadata.
+
+```ts
+core.application.register({
+  id: 'app_status',
+  title: 'App Status',
+  euiIconType: 'snowflake',
+});
+```
+
+This way, the applications could be searchable from the server using this server-side `applications` registry.
+
+However
+
+- It forces plugin developers to add this API call. In addition to be a very poor developer experience, it can also
+  very easily be forgotten, making a given app non searchable
+- client-side only plugins would need to add a server-side part to their plugin just to register their application on
+  the server side
+
```

Comment:
Another hack-ish solution: we can send back to the server all registered on the client-side apps. 🤔 However, it requires that a user lands on a page at least once before calling GS API.

Reply:
Maybe we could hack the chromium instance in Reporting to do this for us? Sounds kinda gross though 😬

Reply:
Main limitation (even with the reporting plugin hack) is that we would have a static list of apps fetched once for all users, which mean that we wouldn't handle apps only registered for specific users/permissions, or things like that.

TBH, I would stick to the client-side providers until we find a clean solution to access apps from the server-side